### PR TITLE
[Validator] [Exception] returns 409 when unique violation is catched

### PR DESCRIPTION
**Pull request by jordscream**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | yes ? 
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #1016 
| License       | MIT
| Doc PR        | N/A

A PR proposal to return 409 Status code when unique violation is catched.

### CODE REVIEW

**Comment by jordscream** [comment]
@dunglas  tell me WYT

**Comment by jordscream** [comment]
@dunglas , it's strange behat tests pass in local, not in travis

File: src/Bridge/Symfony/Validator/Exception/ValidationException.php [hunk 1]
```diff
@@ -38,4 +39,24 @@ public function getConstraintViolationList()
...
+    /**
+     * Check if the list violation only contains Unique Violation.
+     *
+     * @return bool
+     */
+    public function hasOnlyConstraintUniqueViolation()
+    {
+        if (count($this->constraintViolationList) > 0) {
+            foreach ($this->constraintViolationList as $constraintViolation) {
+                if (UniqueEntity::NOT_UNIQUE_ERROR !== $constraintViolation->getCode()) {
```
Comment (dunglas) [inline src/Bridge/Symfony/Validator/Exception/ValidationException.php]: It creates a hard dependency to the Doctrine bridge.

**Comment by dunglas** [comment]
I'm not sure that it's a good idea to implement this in core (it's a very specific use case). Maybe would it better fit in a separate bundle.

**Comment by jordscream** [comment]
no pb :-)

**Pull request closed without merging**